packages/react-router/lib/router/router.ts

```diff
@@ -1366,7 +1371,7 @@ export function createRouter(init: RouterInit): Router {
       return;
     }
 
-    return await startNavigation(historyAction, nextLocation, {
+    await startNavigation(historyAction, nextLocation, {
```

Comment:
Open question - what should happen on interrupted navigations?  Right now the promise will resolve with `undefined` when it gets interrupted.  I don't think rejecting makes sense since we wouldn't want any error boundaries to trigger...